Source/MediaInfo/Multiple/File_DvDif.cpp

```diff
@@ -1626,9 +1626,8 @@ void File_DvDif::video_recdate()
 
     Element_Name("video_recdate");
 
-    Ztring Date=recdate();
     if (Recorded_Date_Date.empty())
-        Recorded_Date_Date=Date;
+        Recorded_Date_Date= recdate();

```

Comment:
Issue is that recdate() is not called in all cases, and it parses a part of the stream so behavior is not exactly same.
But it is true that it is not the best method for parsing, I fixed the issue while changing the way it is implemented, in https://github.com/MediaArea/MediaInfoLib/pull/704.